akka-docs/src/main/paradox/stream/operators/Source-or-Flow/statefulMap.md

```diff
@@ -0,0 +1,35 @@
+# statefulMap
+
+Transform each element into a state and an element that is individually passed downstream.
+
+@ref[Simple operators](../index.md#simple-operators)
+
+## Signature
+
+@apidoc[Flow.statefulMap](Flow) { scala="#statefulMap[S,T](
+create: () = &gt S,f:(S, Out) =&gt (S,T); onComplete: () =$gt Option[T]):FlowOps.this.Repr[T]" java="#statefulMap(
+akka.japi.function.Creator,akka.japi.function.Function,akka.japi.function.Function)" }
+
+## Description
+
+Transform each input element into a pair of an element of T and a state of S. The element will be emitted downstream and
+the state will be passed to the next step in the chain. The returned T MUST NOT be null as it is illegal as stream
+element - according to the Reactive Streams specification. For stateless variant see @ref:[map](map.md). The initial
+state is created using the given function create when the flow is materializing. The onComplete function is called only
+once when the upstream or downstream finished, You can do some clean-up here, and if the returned value is not empty, it
+will be emitted to the downstream if available, otherwise the value will be dropped. Adheres to the
+ActorAttributes.SupervisionStrategy attribute.
+
```

Comment:
This one would be very good to have a few samples for - simple one, one that emits option/optional/sequence and flattens downstream, perhaps one that interacts with some resource, but we can add that later.